Maple2.File.IO/Nif/Endian.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System.Buffers.Binary;
+using System.Numerics;
+using System.Text;
+
+namespace Maple2.File.IO.Nif;
+
+public class EndianReader {
```

Comment:
What about directly using [BinaryReader](https://learn.microsoft.com/en-us/dotnet/api/system.io.binaryreader?view=net-8.0)?

If you also need BigEndian, I think some simple wrappers using extension functions is enough.

Reply:
It's moreso the ability to pick little or big endian at the start of parsing that's needed here because the NIF format allows endianness to be specified by the file. The code would end up looking more or less the same with BinaryReader because it'd still need to store the state of the incoming endianness.